tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -590,6 +590,25 @@ public:
    std::string GetActionName() { return "Take"; }
 };
 
+// Extern templates for TakeHelper
+// NOTE: Due to a compiler bug (?) we declare the move constructor out-of-line as default.
+template <typename RealT_t, typename T, typename COLL>
+TakeHelper<RealT_t, T, COLL>::TakeHelper(TakeHelper<RealT_t, T, COLL> &&) = default;
+template <typename RealT_t, typename T>
+TakeHelper<RealT_t, T, std::vector<T>>::TakeHelper(TakeHelper<RealT_t, T, std::vector<T>> &&) = default;
+template <typename RealT_t, typename COLL>
+TakeHelper<RealT_t, RVec<RealT_t>, COLL>::TakeHelper(TakeHelper<RealT_t, RVec<RealT_t>, COLL> &&) = default;
+template <typename RealT_t>
+TakeHelper<RealT_t, RVec<RealT_t>, std::vector<RealT_t>>::TakeHelper(TakeHelper<RealT_t, RVec<RealT_t>, std::vector<RealT_t>> &&) = default;
+
+extern template class TakeHelper<unsigned int, unsigned int, std::vector<unsigned int>>;
+extern template class TakeHelper<int, int, std::vector<int>>;
+extern template class TakeHelper<unsigned long, unsigned long, std::vector<unsigned long>>;
+extern template class TakeHelper<long, long, std::vector<long>>;
+extern template class TakeHelper<float, float, std::vector<float>>;
+extern template class TakeHelper<double, double, std::vector<double>>;
+
+
```

Comment:
can you remind me the problem of gcc5? abi issues?

Reply:
gcc5 wrongly omits the c++11 abi attribute from certain function signatures, and then cling does not find the symbol (because it looks for the symbol _with_ the attribute).

I think this is the reason for the test failures that we see here, and the only viable solution i know of is to not have the extern templates for gcc5.

Reply:
There is also the clang + cpp17 issue: see ad141836d8f6